src/components/SavedRecipes.tsx

```diff
@@ -21,6 +26,34 @@ const SavedRecipes = () => {
       });
   }, []);
 
+  const handleRecipeSearch = useCallback((params: string) => {
+    const searchedRecipes = recipes.filter((recipe: SavedRecipe) => {
+        const search = params.toLowerCase();
+        const nameSearch = recipe.recipeName.toLowerCase();
+        const ingredientSearch = recipe.recipeIngredients.map((ingredient: { ingredientName: string }) => {
+            return ingredient.ingredientName.toLowerCase();
+        })
+        const tagSearch = recipe.recipeTags.map((tag: { tagName: string; }) => {
+            return tag.tagName.toLowerCase();
+        })
+        return nameSearch.includes(search) || ingredientSearch.includes(search) || tagSearch.includes(search);
+    })
+    if (searchedRecipes.length > 0) {
+      setFilteredRecipes(searchedRecipes);
+      setShow(true);
+    } else if (!searchedRecipes.length) {
+      setShow(false);
+    }
```

Comment:
NON-BLOCKING: I would probably just move this logic to the `return` statement and remove this `show` variable unless you plan on having there be more complex logic than just the length be used to determine whether or not to show the recipes.

Reply:
I can't get it, sorry... If we remove the show variable, what will the conditional rendering be based on?